cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/join/JoinCollection.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.spark.join;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.spark.SparkCollection;
+
+import java.util.List;
+
+/**
+ * Data to join.
+ */
+public class JoinCollection {
+  private final String stage;
+  private final String type;
+  private final SparkCollection<?> data;
+  private final Schema schema;
+  private final List<String> key;
+  private final boolean broadcast;
```

Comment:
Is this used anywhere? The getter doesn't seem to get called. Also it is good to add some comment on what this means.

Reply:
not currently, was going to implement it later.